src/transformers/models/pegasus_x/modeling_pegasus_x.py

```diff
@@ -1299,6 +1300,8 @@ def forward(
         # embed positions
         positions = self.embed_positions(inputs_embeds, past_key_values_length)
 
+        positions = positions.to(inputs_embeds.device)
+
```

Comment:
It is necessary ? I see that in the forward of the `self.embed_positions`, the output tensor is already on the device of `inputs_embeds`. Other than that, LGMT !